lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -1005,15 +1074,27 @@ def _update_label(self, renderer):
         if not self.label.get_visible():
             return
 
+        # We calculate the pad size for the axislabel.
         if self._ticklabel_add_angle != self._axislabel_add_angle:
-            if ((self.major_ticks.get_visible()
-                 and not self.major_ticks.get_tick_out())
-                or (self.minor_ticks.get_visible()
-                    and not self.major_ticks.get_tick_out())):
-                axislabel_pad = self.major_ticks._ticksize
-            else:
-                axislabel_pad = 0
+            # If ticklabels and axislabel are on differenct side, we only
+            # conside the padding for the ticks only.
+
+            ticksizes = []
```

Comment:
Can you add a test that uses this path? (And ideally "all" the newly introduced options.)

Reply:
Yes, I will do.